db/fixtures/ae_datastore/ManageIQ/Deployment/ContainerProvider/System/StateMachine/Deployment.class/__methods__/pre_deployment.rb

```diff
@@ -1,93 +1,70 @@
-require 'net/ssh'
-
 LOCAL_BOOK = 'local_book.yaml'.freeze
-REPO_URL = "https://copr.fedorainfracloud.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo".freeze
+REPO_URL   = "https://copr.fedorainfracloud.org/coprs/maxamillion/origin-next/repo/epel-7/maxamillion-origin-next-epel-7.repo".freeze
 
 def handle_rhel_subscriptions(commands)
-  system "scp -o 'StrictHostKeyChecking no' rhel_subscribe_inventory.yaml " \
-    "#{$evm.root['user']}@#{$evm.root['deployment_master']}:~/"
-  rhel_subscribe_cmd = "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
-                       "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml"
-  commands.push("sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/", rhel_subscribe_cmd)
-end
-
-def ssh_exec!(ssh, command)
-  stdout_data, stderr_data = "", ""
-  exit_code, exit_signal = nil, nil
-
-  ssh.open_channel do |channel|
-    channel.exec("ssh -A -o 'StrictHostKeyChecking no' -t -t #{$evm.root['user']}@#{$evm.root['deployment_master']} " \
-                 + command) do |_, success|
-      raise StandardError, "Command \"#{command}\" was unable to execute" unless success
+  commands.unshift("sudo subscription-manager repos --disable=\"*\"",
+                   "sudo subscription-manager repos --enable=\"rhel-7-server-rh-common-rpms\" --enable=\"rhel-7-server-rpms\" --enable=\"rhel-7-server-extras-rpms\" --enable=\"rhel-7-server-ose-3.2-rpms\"")
+  $evm.root['container_deployment'].perform_scp($evm.root['deployment_master'], $evm.get_state_var(:ssh_user), "rhel_subscribe_inventory.yaml", "rhel_subscribe_inventory.yaml")
```

Comment:
This line is an immediately executed side effect, while rest of this functions just generates commands to be executed later, right?
Could you avoid that and write things in the order they'll be executed?  See comment below in `pre_deployment`, I suspect if you had a non-verbose helper to execute commands, you could just run them here.
I see this function adds commands before *and* after given list.  But maybe it could be 2 functions, or one function taking a block?